app/models/facility.rb

```diff
@@ -101,6 +101,31 @@ class Facility < ApplicationRecord
   delegate :organization, :organization_id, to: :facility_group, allow_nil: true
   delegate :follow_ups_by_period, to: :patients, prefix: :patient
 
+  # ----------------
+  # Region callbacks
+  after_create :create_region, if: -> { Flipper.enabled?(:regions_prep) }
+  after_update :update_region, if: -> { Flipper.enabled?(:regions_prep) }
+
+  def create_region
+    Region.create!(
+      name: name,
+      source: self,
+      reparent_to: block_region,
+      region_type: Region.region_types[:facility]
+    )
+  end
```

Comment:
We talked about this in our zoom sync, but I don't think we should try to create regions for a Facility in a callback...they will be created either:

* via the backfill
* or via the controllers that are directly responsible for mutating Regions

This introduces too many ways that facility/region state can change out from under us in surprising ways.

Reply:
I think it's OK to not do parent regions like Block through the callbacks (as status quo), but I feel really uncomfortable doing concrete data integrity procedures via the API code-paths alone.

We need a robust, single place, that ensures that Regions are always created for source Regions when they are created or edited. The backfill script does not serve this purpose.

I would like to be able to say: _I have a created a facility (regardless of the medium) and I'm guaranteed to have a Region for it, no matter what_.